src/System.Management.Automation/help/ManpageHelpProvider.cs

```diff
@@ -0,0 +1,160 @@
+                manProc.StartInfo.UseShellExecute = false;
+                manProc.StartInfo.RedirectStandardInput = false;
+                manProc.StartInfo.RedirectStandardOutput = true;
+                manProc.StartInfo.RedirectStandardError = false;
+                manProc.StartInfo.UseShellExecute = false;
+                manProc.StartInfo.CreateNoWindow = true;
+                manProc.StartInfo.FileName = "whatis";
+                manProc.StartInfo.Arguments = "-w " + pattern;
+
+                manProc.Start();
+                while (!manProc.StandardOutput.EndOfStream)
+                {
+                    string? line = manProc.StandardOutput.ReadLine();
+                    if (!string.IsNullOrEmpty(line) && line.Trim().Length > 0)
+                    {
+                        // `whatis` line is: "name (section)  - short deswcription"
+                        string[] lineParts = line.Trim().Split(" -", 2);
+
+                        // Determine "name" and "section" from "name (section)"
+                        char[] delim = {'(', ')'};
+                        string[] nameParts = lineParts[0].Trim().Split(delim);
+                        string? manSectionNum = null;
+                        if (nameParts.Length == 1)
+                        {
+                            manSectionNum = string.Empty;   // No section - assume execultable
+                        }
+                        else if (nameParts.Length > 1)
+                        {
+                            string sec = nameParts[1].Trim();
+                            if (sec.Length > 0)
+                            {
+                                string execManSections = "168"; // Man executables sections
+                                if (execManSections.Contains(sec[0]))
+                                {
+                                    manSectionNum = sec;
+                                }
+                            }
+                        }
+
+                        // Add only exceutables to the list
+                        if (manSectionNum != null)
+                        {
+                            string commandName = nameParts[0].Trim();
+                            string commandShortDescription = string.Empty;
+                            if (lineParts.Length > 1)
+                            {
+                                commandShortDescription = lineParts[1].Trim();
+                            }
+
+                            ManpageInfo manpageInfo = new ManpageInfo(commandName, manSectionNum, commandShortDescription);
+                            matchingManpageInfo.Add(manpageInfo);
+                        }
+                    }
+                }
+            }
+            catch (Exception)
+            {
+                // If getting help from man pages failed, do nothing
+                // ????? Should write error using WriteLog()? Similar to Cmdlet.WriteError()? ??????
```

Comment:
?

Reply:
I think that an error message should be logged in this case.  However, I didn't find an available method for error message logging that can be used in this class and I don't know how to develop such method myself.

Reply:
See in other Help providers
`private static readonly PSTraceSource s_tracer = PSTraceSource.GetTracer`

Reply:
Added `PSTraceSource.Writeline` error message.